src/fparser/two/tests/test_comments_and_directives.py

```diff
@@ -514,7 +517,9 @@ def test_all_directive_formats(directive, expected, free):
         source = source + directive + "\n"
         source = source + "        end program foo"
 
-    reader = get_reader(source, isfree=free, ignore_comments=False)
+    reader = get_reader(
```

Comment:
Please duplicate this bit and test that it is sufficient just to supply `process_directives=True` *without* setting `ignore_comments` (with an associated comment).